internal/controllers/operator_controller_test.go

```diff
@@ -1018,6 +1023,94 @@ var _ = Describe("Operator Controller Test", func() {
 			Expect(cond.Message).To(Equal("installation has not been attempted as spec is invalid"))
 		})
 	})
+	When("a catalog changes on cluster", func() {
+		var testLogs, opNames []string
+		var cancel context.CancelFunc
+		var logCount int
+		BeforeEach(func() {
+			l := funcr.New(func(prefix, args string) {
+				if prefix == "operator-controller" &&
+					strings.Contains(args, `"controller"="operator"`) &&
+					strings.Contains(args, `"msg"="ending"`) {
+					// filter for only relevant logs
+					testLogs = append(testLogs, fmt.Sprintf("%s", args))
+				}
+			}, funcr.Options{Verbosity: 1})
+			mgr, err := ctrl.NewManager(cfg, ctrl.Options{Scheme: sch, Logger: l})
+			Expect(err).To(BeNil())
+
+			err = reconciler.SetupWithManager(mgr)
+			Expect(err).To(BeNil())
+			var mgrCtx context.Context
+			mgrCtx, cancel = context.WithCancel(log.IntoContext(ctx, l))
+
+			go func() {
+				err := mgr.Start(mgrCtx)
+				Expect(err).To(BeNil())
+			}()
+
+			opNames = []string{"prometheus", "project-quay"}
+			for _, p := range opNames {
+				op := &operatorsv1alpha1.Operator{ObjectMeta: metav1.ObjectMeta{Name: p}, Spec: operatorsv1alpha1.OperatorSpec{PackageName: p}}
+				err := cl.Create(ctx, op)
+				Expect(err).To(BeNil())
+			}
+			Eventually(func(g Gomega) {
+				By("verifying initial reconcile logs for operator creation")
+				g.Expect(len(testLogs) >= len(opNames)).To(BeTrue())
+				for _, p := range opNames {
+					g.Expect(testLogs[len(testLogs)-len(opNames):]).To(ContainElement(ContainSubstring(fmt.Sprintf("\"Operator\"={\"name\":\"%s\"}", p))))
+				}
+				logCount = len(testLogs)
+			}).WithTimeout(2 * time.Second).WithPolling(1 * time.Second).WithArguments().Should(Succeed())
```

Comment:
Are these custom timeouts and polling configurations necessary?

Reply:
The timeouts are necessary. The default 1s timeout isn't enough time for the operatorReconciler to consistently go through all queued objects, we'd get a lot of flakes without it.

The timeouts are less necessary when switching to a fake reconciler though.